src/Common/src/Interop/Unix/System.Native/Interop.GetCwd.cs

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class Sys
+    {
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        private unsafe delegate void FoundWorkingDirctory(byte* bufferStart);
+
+        [DllImport(Libraries.SystemNative, SetLastError = true)]
+        private static extern void GetCwd(FoundWorkingDirctory foundWorkingDirectory);
+
+        internal unsafe static string GetCwd()
+        {
+            string cwd = string.Empty;
+            GetCwd((byte* bufferStart) => cwd = Marshal.PtrToStringAnsi((IntPtr)bufferStart));
+            return cwd;
```

Comment:
This has two managed heap allocations: one for the delegate and another for the closure object. It'd be nice if there was a fast path that avoided unnecessary managed allocations, like the previous managed stackalloc approach.